### PR TITLE
Start working on the New Game screen

### PR DESCRIPTION
**Pull request by Jummit**
Working on #16.

### CODE REVIEW

**Comment by YeldhamDev** [comment]
Just a tip: Start the pull's title with "WIP:" or "[WIP]" while it's still being worked on, so we can quickly see when it's ready.

**Comment by Jummit** [comment]
> Just a tip: Start the pull's title with "WIP:" or "[WIP]" while it's still being worked on, so we can quickly see when it's ready.

Forgot that.

**Comment by YeldhamDev** [comment]
You'll also have to squish this and your[ other pull](34)'s commits.

**Comment by Jummit** [comment]
> You'll also have to squish this and your[ other pull](34)'s commits.

How do I do that? Is it [this](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)?

**Comment by YeldhamDev** [comment]
Yep, turn those multiple commits into one.

**Comment by Jummit** [comment]
Whats the reason to do so?

**Comment by YeldhamDev** [comment]
A cleaner git history. Besides making much more sense to group together commits that are about a specific change, instead of being scattered around.
Looking at the commits, the main change here is turning the main menu buttons into their own scenes, and as far we care about, this is the only commit message we want.

**Comment by Jummit** [comment]
Superseded by this [pull request](https://github.com/unknown-horizons/godot-port/pull/37).

**Comment by YeldhamDev** [comment]
Apologies for not checking this pull sooner, should've tried to merge this first before upgrading the project to 3.1. Could you rebase this one more time?

**Comment by Jummit** [comment]
> Apologies for not checking this pull sooner, should've tried to merge this first before upgrading the project to 3.1. Could you rebase this one more time?

Done.

**Comment by YeldhamDev** [comment]
There's still a small conflict in the `icon.png.import` file.

**Comment by olivier-grech** [comment]
I have a small suggestion. Currently, the checkbox for the game type (**Scenario**, **Random map**, **Free play**) can be unchecked by clicking on it.

I think it should behave like a set of radio buttons: one of the cases is checked at first (let's say **Scenario**), and when you click on something that is already checked it does nothing (therefore there is alway one case checked).

Here is an example of how you could do that for the `GameTypeButton.gd` script:

```GDScript
extends HBoxContainer

export var type = ""
export var checked = false

func _ready():
	$Label.text = type
	if checked:
		get_node("CheckBox").pressed = true


func _on_pressed():
	for type_button in get_parent().get_children():
		if type_button != self:
			type_button.get_node("CheckBox").pressed = false
	if !get_node("CheckBox").pressed:
		get_node("CheckBox").pressed = true
```

**Comment by olivier-grech** [comment]
Another small suggestion, I think you should indicate the number of AI players selected, this is not very clear right now. Something like this:

exampe

Though I am not sure if the leftmost option should be `1` or `0`.

**Comment by aaronfranke** [comment]
Not only will you need to rebase, but you will also need to remove some files - I added `.DS_Store` to the `.gitignore` file.

**Comment by LinuxDonald** [comment]
It should be 0 if anyone want to play without ai.

File: Assets/UI/Scenes/MainMenuScene.tscn [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [ext_resource path="res://Assets/UI/Images/Background/MainMenu/bg_1.png" type="Texture" id=1]
 [ext_resource path="res://Assets/UI/Scenes/MainMenuUI.tscn" type="PackedScene" id=2]
-[ext_resource path="res://Assets/Audio/Music/Menu/main_theme.ogg" type="AudioStream" id=3]
+[ext_resource path="res://Assets/UI/Images/Background/Widgets/main_theme.ogg" type="AudioStream" id=3]
```
Comment (aaronfranke) [inline Assets/UI/Scenes/MainMenuScene.tscn:5]: This file doesn't exist. This change breaks the menu music.

**Review by aaronfranke — CHANGES_REQUESTED** [review]
You need to rebase your commit using [a Git client](https://www.gitkraken.com/). Please see [PR workflow](https://godot.readthedocs.io/en/latest/community/contributing/pr_workflow.html).

**Comment by LinuxDonald** [comment]
@Jummit are you still working on this?

**Comment by Jummit** [comment]
> @Jummit are you still working on this?

I sadly can't right now, because I have RSI and don't want it to get worse.

**Comment by aaronfranke** [comment]
Superseded by https://github.com/unknown-horizons/godot-port/pull/43

**Pull request closed without merging**